File: Core/Team/src/main/java/org/team/application/commands/RemoveTeammateFromTeam.java
package org.team.application.commands;

import org.shared.api.Command;
import org.team.entities.Team;
import org.team.ports.dto.RemoveTeammateFromTeamRequest;
import org.team.ports.spi.TeamRepository;

public class RemoveTeammateFromTeam implements Command<RemoveTeammateFromTeamRequest> {
  private final TeamRepository teamRepository;

  public RemoveTeammateFromTeam(TeamRepository teamRepository) {
    this.teamRepository = teamRepository;
  }


  public void execute(RemoveTeammateFromTeamRequest request) {
    var foundedTeam = teamRepository.getTeamById(request.teamId()).orElseThrow(()-> new RuntimeException("not found"));
    var team = new Team(foundedTeam.id(), foundedTeam.name(), foundedTeam.teammates(), foundedTeam.accountId());
    team.removeTeammate(request.teammateId());
    teamRepository.saveTeam(team.snapshot());
  }
}

File: frontend/src/core/application/team/command/RemoveTeammateFromTeam.ts
import { Command } from "src/core/application/shared/command/Command";
import { TeamCommandHandler } from "../spi/TeamCommandHandler";
import { RemoveTeammateFromTeamRequest } from "../dto/RemoveTeammateFromTeamRequest";
import { Observable, catchError, map } from "rxjs";
import { Result } from "src/core/application/shared/dto/Result";




export class RemoveTeammateFromTeam extends Command<string, RemoveTeammateFromTeamRequest> {

    public constructor(private commandHandler: TeamCommandHandler){super()}

    public execute(request: RemoveTeammateFromTeamRequest): Observable<Result<string>> {
        return this.commandHandler.removeTeammateFromTeam(request)
        .pipe(map(this.onSuccess),
         catchError(this.onError))
    }
}